File: C#/Arrays/Exercism - Bird Watcher.cs
// Link to the exercise: https://exercism.org/tracks/csharp/exercises/bird-watcher

using System;

class BirdCount
{
    private int[] birdsPerDay;
    private static int[] _birdsLastWeek = { 0, 2, 5, 3, 7, 8, 4 };

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek() =>
        _birdsLastWeek;

    public int Today() =>
        birdsPerDay[^1];

    public void IncrementTodaysCount() =>
        birdsPerDay[^1]++;

    public bool HasDayWithoutBirds()
    {
        foreach (int day in birdsPerDay)
        {
            if (day == 0)
                return true;
        }
        return false;
    }

    public int CountForFirstDays(int numberOfDays)
    {
        int counter = 0;

        for (int i = 0; i < numberOfDays; i++)
        {
            counter += birdsPerDay[i];
        }

        return counter;
    }

    public int BusyDays()
    {
        int busyCounter = 0;
        foreach (int day in birdsPerDay)
        {
            if (day >= 5)
                busyCounter++;
        }
        return busyCounter;
    }
}

File: Java/Arrays/Exercism - Bird Watcher.java
// Link to the exercise: https://exercism.org/tracks/java/exercises/bird-watcher

public class BirdWatcher {
    private final int[] birdsPerDay;
    private static int[] birdsLastWeek = { 0, 2, 5, 3, 7, 8, 4 };

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() 
    {
        return birdsLastWeek;
    }

    public int getToday() 
    {
        return birdsPerDay[(birdsPerDay.length - 1)];
    }

    public void incrementTodaysCount() 
    {
        birdsPerDay[(birdsPerDay.length - 1)]++;
    }

    public boolean hasDayWithoutBirds() 
    {
        for (int day: birdsPerDay) 
        {
            if (day == 0)
                return true;
        }

        return false;
    }

    public int getCountForFirstDays(int numberOfDays) 
    {
        int counter = 0;

        for (int i = 0; i < numberOfDays; i++) 
        {
            if (i <= 6)
                counter += birdsPerDay[i];
        }

        return counter;
    }

    public int getBusyDays() 
    {
        int busyCounter = 0;

        for (int day: birdsPerDay)
        {
            if (day >= 5)
                busyCounter++;
        }

        return busyCounter;
    }
}
